homeassistant/components/nexia/config_flow.py

```diff
@@ -0,0 +1,92 @@
+"""Config flow for Nexia integration."""
+from functools import partial
+import logging
+
+from nexia.home import NexiaHome
+from requests.exceptions import ConnectTimeout, HTTPError
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+
+from .const import DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema({CONF_USERNAME: str, CONF_PASSWORD: str})
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+    nexia_home = None
+    try:
+        nexia_home = await hass.async_add_executor_job(
+            partial(
+                NexiaHome,
+                username=data[CONF_USERNAME],
+                password=data[CONF_PASSWORD],
+                auto_login=False,
+                auto_update=False,
+            )
+        )
+        nexia_home.login()
```

Comment:
Is this doing I/O ? We're in an async function.

Reply:
Yes it is .. *doh*

Fixing

Reply:
fixed